src/main/java/teammates/ui/webapi/action/GetCoursesAction.java

```diff
@@ -17,17 +24,73 @@ protected AuthType getMinAuthLevel() {
 
     @Override
     public void checkSpecificAccessControl() {
-        gateKeeper.verifyInstructorPrivileges(logic.getAccount(userInfo.getId()));
+        String entityType = getNonNullRequestParamValue(Const.ParamsNames.ENTITY_TYPE);
+
+        if (!((entityType.equals(Const.EntityType.STUDENT) && userInfo.isStudent)
+                || (entityType.equals(Const.EntityType.INSTRUCTOR) && userInfo.isInstructor))) {
+            throw new UnauthorizedAccessException("Current account cannot access to courses of request entity type");
+        }
     }
 
     @Override
     public ActionResult execute() {
-        List<CourseAttributes> courses =
-                logic.getCoursesForInstructor(logic.getInstructorsForGoogleId(userInfo.getId(), true));
+        String entityType = getNonNullRequestParamValue(Const.ParamsNames.ENTITY_TYPE);
+        switch (entityType) {
+        case Const.EntityType.STUDENT:
+            return getStudentCourses();
+        case Const.EntityType.INSTRUCTOR:
+            return getInstructorCourses();
+        default:
+            return new JsonResult("Error: invalid entity type", HttpStatus.SC_BAD_REQUEST);
```

Comment:
Redundant as it will never reach this case given you already checked in `checkSpecificAccessControl()` ?

Reply:
Yes, but I believe this is a good practice to use `default` for `switch` even if it is impossible to happen